tests/Integration/Behaviour/Features/Context/CartRuleFeatureContext.php

```diff
@@ -118,13 +120,22 @@ public function thereIsACartRuleWithNameAndAmountDiscountOfAndPriorityOfAndQuant
         $this->createCartRule($cartRuleName, 0, $amount, $priority, $cartRuleQuantity, $cartRuleQuantityPerUser);
     }
 
+    /**
+     * @Given /^there is a cart rule named "(.+)" with empty code that applies an amount discount of (\d+\.\d+) with priority (\d+), quantity of (\d+) and quantity per user (\d+)$/
+     */
+    public function thereIsACartRuleWithEmptyCodeAndNameAndAmountDiscountOfAndPriorityOfAndQuantityOfAndQuantityPerUserOf($cartRuleName, $amount, $priority, $cartRuleQuantity, $cartRuleQuantityPerUser)
+    {
+        $this->createCartRule($cartRuleName, 0, $amount, $priority, $cartRuleQuantity, $cartRuleQuantityPerUser, '');
+    }
+
     protected function createCartRule(
         $cartRuleName,
         $percent,
         $amount,
         $priority,
         $cartRuleQuantity,
-        $cartRuleQuantityPerUser
+        $cartRuleQuantityPerUser,
+        $code = null
```

Comment:
I don't understand the use of this extra `$code` parameter since you don't use in your endpoint any way

Ok it's empty string instead of null, but does it really changes the behavior? That would mean all present cart rules have been created with a code and we didn't realize it?